apps/src/p5lab/spritelab/libraries/PoemBotLibrary.js

```diff
@@ -5,62 +5,58 @@ import {commands as foregroundEffects} from '../commands/poembot/foregroundEffec
 const OUTER_MARGIN = 50;
 const LINE_HEIGHT = 50;
 const FONT_SIZE = 25;
-const TextType = {
-  BLANK: 'blank',
-  LITERAL: 'literal',
-  RANDOM: 'random'
-};
-const BLANK_TEXT = {value: '', type: TextType.BLANK};
 const PLAYSPACE_SIZE = 400;
+const POEM_DURATION = 500;
 
 export default class PoemBotLibrary extends CoreLibrary {
   constructor(p5) {
     super(p5);
-    this.animationSpeed = 100; // time in frames
-    this.currentLine = 0;
-    this.poem = {
+    this.poemState = {
       title: '',
       author: '',
-      lines: []
+      lines: [],
+      color: 'black',
+      font: 'Arial',
+      isVisible: true,
+      effects: []
     };
-    this.isVisible = false;
     this.backgroundEffect = () => this.p5.background('white');
     this.foregroundEffect = () => {};
     this.lineEvents = {};
     this.p5.noStroke();
+    this.p5.textAlign(this.p5.CENTER);
 
     this.commands = {
       // Keep everything from Core Sprite Lab
       ...this.commands,
 
       // Override the draw loop
       executeDrawLoopAndCallbacks() {
-        if (
-          this.p5.World.frameCount % this.animationSpeed === 0 &&
-          this.poem.lines.length > this.currentLine
-        ) {
-          this.currentLine++;
-
-          // Call callbacks for any line events at the current line
-          this.lineEvents[this.currentLine]?.forEach(callback => callback());
-        }
         this.backgroundEffect();
         this.runBehaviors();
         this.runEvents();
         this.p5.drawSprites();
-        this.drawPoem();
+        const renderInfo = this.getRenderInfo(
+          this.poemState,
+          this.p5.World.frameCount
```

Comment:
Does this mean we can scrub time forwards and backwards? :)

Reply:
I think so, though we probably won't have a way to actually support that in the UI in spritelab any time soon. It should make unit testing easier though since `getRenderInfo` is a pure function.